### DISCUSSION TITLE
Azure AD Authentication Issue

### DISCUSSION THREAD

**Discussion by deeepakkumarthakur16** [requarks/wiki#5865] [Error / Bug Report]
I have integrated my wikijs with azure ad. I spent lot of time to make it work as i follow the official documentation correctly but i was getting error. **Oops, something went wrong...Invalid email / username or password.** Later on i notice i got the error but my azure ad user appeared in wikijs users list then i try to login in incognito mode and able to log in like charm..... but if i logout the user and try to login again got same error. After this, i got it that wikijs not working properly with existing cookies....

@NGPixel is there any solution or fix do we have for this?

On relogin callback url throwing 500 internal server error.
<img width="680" alt="image" src="https://user-images.githubusercontent.com/64971445/202726085-74ae481e-ccec-4108-aa66-eaa0616b6e2a.png">

**Comment by SimonCoutureCSTJ** [comment]
Same issue here. 

@NGPixel  Any idea? I would really like this feature.

**Comment by rrg92** [comment]
Hello,

I suspect the issue you're experiencing with Wiki.js and Azure authentication may stem from SameSite cookie restrictions. The Azure authentication module in Wiki.js uses an implicit flow that relies on cookies for maintaining session state.

During your initial login, you're redirected to the Microsoft site where you're presented with a login form. Upon successful authentication, you're redirected back to Wiki.js. During this process, since the redirection originates directly from a user action on the Microsoft page, cookies are successfully transferred without any issues.

However, if you log out of Wiki.js and attempt to log in again, Microsoft doesn't present the login form if it detects you're still logged in. Instead, it automatically redirects you back to Wiki.js using JavaScript to submit a form (which includes sending tokens via form post). This automated redirection, lacking direct user interaction, may prevent cookies from being properly sent to the Wiki.js domain due to SameSite policy enforcement. These cookies are crucial as they contain session information and other vital authentication data. The absence of these cookies leads to authentication errors, resulting in Wiki.js displaying a generic 'invalid user' message.

To address this, I plan to submit a pull request that will improve the authentication flow by implementing the Authorization Code Flow. This approach does not rely on cookies, thus bypassing the issues caused by SameSite restrictions.

As a temporary workaround, consider using the OIDC authentication module. You'll need to configure each field manually. Make sure to use the "username" field to capture the email claim. This should help mitigate the problem until a more permanent solution is implemented.
